internal/issuer/clients/cert/requests_test.go

```diff
@@ -0,0 +1,155 @@
+package cert
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var exampleBytes = []byte("example")
+
+const (
+	PostCert         = "PostCert"
+	GetCert          = "GetCert"
+	FailGetCert      = "FailGetCert"
+	InvalidResponse  = "InvalidResponse"
+	TestURL          = "https://test.com/"
+	DownloadEndpoint = "download/"
+	GUID             = "12345678/"
+)
+
+func TestRequests(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+	ctx := context.Background()
+	log := zap.New()
+	type args struct {
+		name   string
+		client Client
+	}
+	type want struct {
+		method    string
+		path      string
+		responder httpmock.Responder
+	}
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"ShouldSendPostRequest": {
+			args: args{
+				name:   PostCert,
+				client: NewClient(WithAPIEndpoint(TestURL), WithHTTPClient(hClient)),
+			},
+			want: want{
+				method: http.MethodPost,
+				path:   TestURL + "csr",
+				responder: func(request *http.Request) (*http.Response, error) {
+					requestBody, _ := io.ReadAll(request.Body)
+					if !bytes.Equal(requestBody, exampleBytes) {
+						return httpmock.NewJsonResponse(http.StatusBadRequest, fmt.Sprintf("got %b, want %b", request.Body, exampleBytes))
+					}
```

Comment:
Why is this needed?

Reply:
testing to see it sends the certificate correctly